packages/core/content-manager/shared/contracts/collection-types.ts

```diff
@@ -170,12 +170,34 @@ export declare namespace Delete {
   }
 }
 
+/**
+ * POST /collection-types/:model/actions/publish
+ */
+export declare namespace PublishAndCreate {
+  export interface Request {
+    body: Document;
+    query: {
+      locale?: string | null;
+    };
+  }
+
+  export interface Params {
+    model: string;
+  }
+
+  export interface Response {
+    data: Document;
+    meta: DocumentMetadata;
+    error?: errors.ApplicationError;
+  }
+}
+
 /**
  * POST /collection-types/:model/:id/actions/publish
```

Comment:
this should work for single-types too right?

Reply:
Yep, I was just working on the logic of it too ! Will be ready before merging 🫡

Reply:
I will have a happier life the day we unify the single type/ collection type logic

Reply:
If you're speedy enough we can do it post public beta 😉

Reply:
<img src="https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExNjlnaXltODRkbm9qcDV2NG1teTNuM3ExZnBwaTh2ZnR1a2l3YWZudyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/xTiTntdVHPOnIhK608/giphy.gif"/>